src/images.ts

```diff
@@ -0,0 +1,32 @@
+import type { Root } from 'mdast';
+import type { StaticNotebook } from '@jupyterlab/notebook';
+import type { Image } from 'myst-spec';
+import { selectAll } from 'unist-util-select';
+
+type Options = {
+  parent: StaticNotebook;
+};
+
+export async function imageUrlSourceTransform(
+  tree: Root,
+  opts: Options
+): Promise<void> {
+  const images = selectAll('image', tree) as Image[];
+  await Promise.all(
+    images.map(async image => {
+      if (!image || !image.url) return;
+      // TODO: not sure why, but the cell seems to have a private _rendermime?
+      // How else to get `attachment:` to work?
+      const rendermime =
+        opts.parent.rendermime ??
+        (opts.parent as any)._rendermime ??
+        (opts.parent.parent as any).rendermime;
```

Comment:
I could use some help understanding this, the cell seems to have the information for the rendermime image replacement, but that is private (not sure why!?).